.github/workflows/build-vault-oss.yml

```diff
@@ -40,18 +49,31 @@ jobs:
     runs-on: ubuntu-latest
     name: Vault ${{ inputs.goos }} ${{ inputs.goarch }} v${{ inputs.vault-version }}
     steps:
-      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
-      - uses: actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9 # v4.0.0
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
         with:
           go-version: ${{ inputs.go-version }}
-      - name: Set up node and yarn
-        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+          cache: false # Use our own caching strategy for better cross platform support
+      - name: Set up Go module cache
+        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
+        with:
+          path: |
+            ${{ inputs.go-cache }}
+            ${{ inputs.go-mod-cache }}
+          # Manage the cache for each platform and architecture combo. In this way we'll only ever
+          # restore modules and build cache for the platform and architecture we're targeting.
+          key: ${{ inputs.go-version }}-${{ inputs.goos }}-${{ inputs.goarch }}-go-${{ hashFiles('**/go.sum') }}
```

Comment:
We should probably include a hash of `go-tags` in here as well. In the event we build for the same platform/architecture but with different tags, eg: ent, hsm, fips, hsm.fips, we'd want to have separate build caches.